btb/benchmark/__init__.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+
+from btb.benchmark.challenges import Bohachevsky, Branin, Rosenbrock
+
+DEFAULT_CHALLENGES = [
+    Bohachevsky,
+    Branin,
+    Rosenbrock,
+]
+
+
+def benchmark(tuning_functions, challenges=DEFAULT_CHALLENGES, iterations=1000):
+    """Benchmark function.
+
+    This benchmark function iterates over a collection of ``challenges`` and executes a
+    ``tuner_function`` for each one of the ``challenges`` for a given amount of iterations.
+
+    Args:
+        tuning_functions (dict):
```

Comment:
Maybe rename this to `candidates` (or `challengers`) and make this a bit more flexible?

```
if callable(candidates):
    candidates = {candidates.__name__: candidates}
elif not isinstance(candidates, dict):
    raise TypeError('Candidates can only be callable or dict')
```

So one can directly do:

```
benchmark(my_function)
```

Optionally, support for list or tuple can also be added:

```
benchmark((my_function, my_other_function))
```